src/averageAges.js

```diff
@@ -1,61 +1,83 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let men = people.filter(person => person.sex === 'm');
+
+  men = century
+    ? men.filter(person => Math.ceil(person.died / 100) === century)
+    : men;
```

Comment:
You can just do like that)
```suggestion
  const men = people.filter(person => century 
   ? Math.ceil(person.died / 100) === century && person.sex === 'm');
   : person.sex === 'm'
 )
```